### PR TITLE
Add more controller tests

### PR DESCRIPTION
**Pull request by vjuranek**

### CODE REVIEW

**Review by ghost — CHANGES_REQUESTED** [review]
Great work, amazing suite of tests. A few wording improvements and we are good to go.

File: pkg/service/controller_test.go [hunk 1]
```diff
@@ -6,14 +6,279 @@ import (
...
+	})
+	if err != nil {
+		t.Fatalf("failed to create deleteVolumeResponse (%v)", err)
+	}
+	diskList, err := helper.GetClient().ListDisks()
+	if err != nil {
+		t.Fatalf("failed to list disks (%v)", err)
+	}
+	if len(diskList) != 0 {
+		t.Fatalf("incorrect number of disks created (%d)", len(diskList))
```
[the remark this answers is not in the document]
Comment (vjuranek) [reply pkg/service/controller_test.go:55]: done

File: pkg/service/controller_test.go [hunk 2]
```diff
@@ -6,14 +6,279 @@ import (
...
+	}
+}
+
+func TestDeleteNonexistentVolume(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (vjuranek) [reply pkg/service/controller_test.go:59]: done

File: pkg/service/controller_test.go [hunk 3]
```diff
@@ -6,14 +6,279 @@ import (
...
+	helper := getMockHelper(t)
+	controller := service.NewOvirtCSIDriver(helper.GetClient(), "test")
+
+	_, err := controller.DeleteVolume(context.Background(), &csi.DeleteVolumeRequest{
+		VolumeId: "doesn't exists",
+	})
+	if err != nil {
+		t.Fatalf("failed to create deleteVolumeResponse (%v)", err)
```
[the remark this answers is not in the document]
Comment (vjuranek) [reply pkg/service/controller_test.go:67]: Error message was improved.

Yes, it should be idempotent:
```
This operation MUST be idempotent. If a volume corresponding to the specified volume_id does not exist or the artifacts associated with the volume do not exist anymore, the Plugin MUST reply 0 OK.
```
that's why test fails if `err != nil`

File: pkg/service/controller_test.go [hunk 4]
```diff
@@ -6,14 +6,279 @@ import (
...
+				Mode: csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER,
+			},
+		},
+		Readonly: true,
+	}
+
+	// Publish volume.
+	_, _, err := publishTestVolume(helper, publishRequest)
+	if err != nil {
+		t.Fatalf("failed to create controllerPublishVolumeResponse (%v)", err)
```
[the remark this answers is not in the document]
Comment (vjuranek) [reply pkg/service/controller_test.go:135]: done

File: pkg/service/controller_test.go [hunk 5]
```diff
@@ -6,14 +6,279 @@ import (
...
+	}
+
+	// Publish volume again, should succeed.
+	_, _, err = publishTestVolume(helper, publishRequest)
+	if err != nil {
+		t.Fatalf("failed to create controllerPublishVolumeResponse for already published volume (%v)", err)
+	}
+}
+
+func TestControllerUnpublishVolume(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (vjuranek) [reply pkg/service/controller_test.go:145]: done

File: pkg/service/controller_test.go [hunk 6]
```diff
@@ -6,14 +6,279 @@ import (
...
+			},
+			AccessMode: &csi.VolumeCapability_AccessMode{
+				Mode: csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER,
+			},
+		},
+		Readonly: true,
+	}
+	vmId, diskId, err := publishTestVolume(helper, publishRequest)
+	if err != nil {
+		t.Fatalf("failed to create controllerPublishVolumeResponse (%v)", err)
```
[the remark this answers is not in the document]
Comment (vjuranek) [reply pkg/service/controller_test.go:164]: done

File: pkg/service/controller_test.go [hunk 7]
```diff
@@ -6,14 +6,279 @@ import (
...
+			RequiredBytes: 1024,
+			LimitBytes:    1024,
+		},
+	})
+	if err != nil {
+		t.Fatalf("failed to expand volume (%v)", err)
+	}
+
+	if expandResp.CapacityBytes < 4096 {
+		t.Fatalf("volume was shrink (%v)", err)
```
[the remark this answers is not in the document]
Comment (vjuranek) [reply pkg/service/controller_test.go:258]: done

File: pkg/service/controller_test.go [hunk 8]
```diff
@@ -6,14 +6,279 @@ import (
...
+	}
+	if expandResp.NodeExpansionRequired {
+		t.Fatalf("volume was expanded (%v)", err)
```
[the remark this answers is not in the document]
Comment (vjuranek) [reply pkg/service/controller_test.go:261]: done

**Review by vjuranek — COMMENTED** [review]
Error messages were improved, let me know it they need further work (or anything else).

File: pkg/service/controller_test.go [hunk 9]
```diff
@@ -6,14 +6,324 @@ import (
...
+	createVolumeResponse, err := createTestVolume(helper, controller)
+	if err != nil {
+		t.Fatalf("failed to create test volume (%v)", err)
+	}
+
+	_, err = controller.DeleteVolume(context.Background(), &csi.DeleteVolumeRequest{
+		VolumeId: createVolumeResponse.Volume.VolumeId,
+	})
+	if err != nil {
+		t.Fatalf("failed to create deleteVolumeResponse (%v)", err)
```
Comment (janosdebugs) [inline pkg/service/controller_test.go:48]: Sorry to poke, this is also in need of a better error message. Our future self will thank us if this test fails.

**Review by ghost — CHANGES_REQUESTED** [review]
One last error message, then we are good.

File: pkg/service/controller_test.go (hunk 9 above)
Comment (vjuranek) [reply pkg/service/controller_test.go]: np, my fault. Should be fixed now.

**Review by ghost — APPROVED** [review]
/lgtm

**Pull request merged**